File: exemplos/ruby/consulta_ie.rb
# FiscalAPI — Exemplo de consulta de Inscricao Estadual em Ruby
# https://fiscalapi.com.br

require "net/http"
require "uri"
require "json"

API_KEY = ENV.fetch("FISCALAPI_KEY")
BASE_URL = "https://api.fiscalapi.com.br"

def consultar_ie(uf:, cpf: nil, cnpj: nil, ie: nil)
  params = { uf: uf }
  params[:cpf] = cpf if cpf
  params[:cnpj] = cnpj if cnpj
  params[:ie] = ie if ie

  uri = URI("#{BASE_URL}/api/consultar?#{URI.encode_www_form(params)}")
  request = Net::HTTP::Get.new(uri)
  request["X-API-Key"] = API_KEY

  response = Net::HTTP.start(uri.hostname, uri.port, use_ssl: true, read_timeout: 90) do |http|
    http.request(request)
  end

  data = JSON.parse(response.body)
  raise "#{data['error']}: #{data['message']}" unless response.is_a?(Net::HTTPSuccess)

  data
end

def consultar_ie_todos(cpf: nil, cnpj: nil)
  params = {}
  params[:cpf] = cpf if cpf
  params[:cnpj] = cnpj if cnpj

  uri = URI("#{BASE_URL}/api/consultar-ie-todos?#{URI.encode_www_form(params)}")
  request = Net::HTTP::Get.new(uri)
  request["X-API-Key"] = API_KEY

  response = Net::HTTP.start(uri.hostname, uri.port, use_ssl: true, read_timeout: 120) do |http|
    http.request(request)
  end

  data = JSON.parse(response.body)
  raise "#{data['error']}: #{data['message']}" unless response.is_a?(Net::HTTPSuccess)

  data
end

# Exemplo de uso
if __FILE__ == $PROGRAM_NAME
  resultado = consultar_ie(uf: "MT", cpf: "12345678900")
  puts "Encontradas #{resultado['results'].length} inscricoes em MT"

  resultado["results"].each do |ie|
    puts "  IE: #{ie['inscricao_estadual']} — #{ie['situacao_ie']}"
    puts "  Razao: #{ie['razao_social']}"
    puts "  Municipio: #{ie['municipio']}\n\n"
  end
end

File: exemplos/php/consulta_ie.php
<?php
/**
 * FiscalAPI — Exemplo de consulta de Inscricao Estadual em PHP
 * https://fiscalapi.com.br
 */

$apiKey = getenv('FISCALAPI_KEY');
$baseUrl = 'https://api.fiscalapi.com.br';

/**
 * Consulta IE de um produtor rural em um estado especifico.
 */
function consultarIE(string $uf, ?string $cpf = null, ?string $cnpj = null, ?string $ie = null): array
{
    global $apiKey, $baseUrl;

    $params = ['uf' => $uf];
    if ($cpf) $params['cpf'] = $cpf;
    elseif ($cnpj) $params['cnpj'] = $cnpj;
    elseif ($ie) $params['ie'] = $ie;

    $url = $baseUrl . '/api/consultar?' . http_build_query($params);

    $ch = curl_init($url);
    curl_setopt_array($ch, [
        CURLOPT_RETURNTRANSFER => true,
        CURLOPT_TIMEOUT => 90,
        CURLOPT_HTTPHEADER => [
            'X-API-Key: ' . $apiKey,
        ],
    ]);

    $response = curl_exec($ch);
    $httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
    curl_close($ch);

    $data = json_decode($response, true);

    if ($httpCode !== 200) {
        throw new Exception("{$data['error']}: {$data['message']}");
    }

    return $data;
}

/**
 * Consulta IE em todos os estados simultaneamente.
 */
function consultarIETodos(?string $cpf = null, ?string $cnpj = null): array
{
    global $apiKey, $baseUrl;

    $params = [];
    if ($cpf) $params['cpf'] = $cpf;
    elseif ($cnpj) $params['cnpj'] = $cnpj;

    $url = $baseUrl . '/api/consultar-ie-todos?' . http_build_query($params);

    $ch = curl_init($url);
    curl_setopt_array($ch, [
        CURLOPT_RETURNTRANSFER => true,
        CURLOPT_TIMEOUT => 120,
        CURLOPT_HTTPHEADER => [
            'X-API-Key: ' . $apiKey,
        ],
    ]);

    $response = curl_exec($ch);
    $httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
    curl_close($ch);

    $data = json_decode($response, true);

    if ($httpCode !== 200) {
        throw new Exception("{$data['error']}: {$data['message']}");
    }

    return $data;
}

// Exemplo de uso
$resultado = consultarIE('MT', cpf: '12345678900');
echo "Encontradas " . count($resultado['results']) . " inscricoes em MT\n";

foreach ($resultado['results'] as $ie) {
    echo "  IE: {$ie['inscricao_estadual']} — {$ie['situacao_ie']}\n";
    echo "  Razao: {$ie['razao_social']}\n";
    echo "  Municipio: {$ie['municipio']}\n\n";
}
